src/installer.ts

```diff
@@ -76,15 +77,105 @@ export async function getJuliaVersions(versionInfo): Promise<string[]> {
+        range = range.replace("≥", ">=").replace("≤", "<=")
+
+        // Cleanup whitespace. Julia only allows whitespace between the specifier and version with certain specifiers
+        range = range.replace(/\s+/g, " ").replace(/(?<=(>|>=|≥|<)) (?=\d)/g, "")
+
+        if (!semver.validRange(range) || range.split(/(?<! -) (?!- )/).length > 1 || range.startsWith("<=") || range === "*") {
+            return null
+        } else if (range.search(/^\d/) === 0 && !range.includes(" ")) {
+            // Compat version is just a basic version number (e.g. 1.2.3). Since Julia's Pkg.jl's uses caret
+            // as the default specifier (e.g. `1.2.3 == ^1.2.3`) and NPM's semver uses tilde as the default
+            // specifier (e.g. `1.2.3 == 1.2.x == ~1.2.3`) we will introduce the caret specifier to ensure the
+            // orignal intent is respected.
+            // https://pkgdocs.julialang.org/v1/compatibility/#Version-specifier-format
+            // https://github.com/npm/node-semver#x-ranges-12x-1x-12-
+            range = "^" + range
+        }
+
+        ranges.push(range)
+    }
+
+    return semver.validRange(ranges.join(" || "))
+}
+
+/**
+ * @returns An array of version ranges compatible with the Julia project
+ */
+export function readJuliaCompatRange(projectFileContent: string): string {
+    let compatRange: string | null
+    let meta = toml.parse(projectFileContent)
+
+    if (meta.compat?.julia !== undefined) {
+        compatRange = validJuliaCompatRange(meta.compat.julia)
+    } else {
+        compatRange = "*"
+    }
+
+    if (!compatRange) {
+        throw new Error(`Invalid version range found in Julia compat: ${compatRange}`)
+    }
+
+    return compatRange
+}
+
+export function getJuliaVersion(availableReleases: string[], versionInput: string, includePrerelease: boolean = false, juliaCompatRange: string = ""): string {
+    let version: string | null
+
     if (semver.valid(versionInput) == versionInput || versionInput.endsWith('nightly')) {
         // versionInput is a valid version or a nightly version, use it directly
-        return versionInput
+        version = versionInput
+    } else if (versionInput == "MIN") {
+        // Resolve "MIN" to the minimum supported Julia version compatible with the project file
+        if (!juliaCompatRange) {
+            throw new Error('Unable to use version "MIN" when the Julia project file does not specify a compat for Julia')
+        }
+        version = semver.minSatisfying(availableReleases, juliaCompatRange, {includePrerelease})
+    } else {
+        // Use the highest available version that matches versionInput
+        version = semver.maxSatisfying(availableReleases, versionInput, {includePrerelease})
     }
 
-    // Use the highest available version that matches versionInput
-    let version = semver.maxSatisfying(availableReleases, versionInput, {includePrerelease})
-    if (version == null) {
+    if (!version) {
```

Comment:
What's the difference between `(!version)` (this PR) and `if (version == null)` (on master)?

Reply:
So the issue with the original code is that `null == undefined` is also true so the original statement is kind of misleading. I switched things to be based on "truthy"-ness so now we thrown an exception on any value that is coerced into a `false` such as `null`, `undefined`, or `""`.